engine/api/workflow/dao_node_run.go

```diff
@@ -658,7 +658,8 @@ func updateNodeRunCommits(db gorp.SqlExecutor, id int64, commits []sdk.VCSCommit
 	return nil
 }
 
-func updateNodeRunStatusAndStage(db gorp.SqlExecutor, nodeRun *sdk.WorkflowNodeRun) error {
+// UpdateNodeRunStatusAndStage update just noderun status and stage
+func UpdateNodeRunStatusAndStage(db gorp.SqlExecutor, nodeRun *sdk.WorkflowNodeRun) error {
```

Comment:
I think `updateNodeRunStatusAndStage` should not be public, it's a very low-level function, only called by other private functions.

Maybe the goal achieved in the purge package querying and modifying workflowNoRun is also a bit low-level and should be in the workflow package.